### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by Seyi-Keye**
Basically 
> adding test to this project 
> fixing subsequent api calls in lifecycle methods 
> displaying error appropriately.

#### App initial state
screenshot 2018-04-24 00 25 42

#### App state with result displayed
screenshot 2018-04-24 00 26 11

#### App state with error
screenshot 2018-04-24 00 27 12

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,35 +1,39 @@
-import React, { Component } from 'react';
+import React, { Component } from "react";
 import GridView from "./GridView";
 import SearchTool from "./Searchtool";
-import './App.css';
+import "./App.css";
```
Comment (abiodun0) [inline src/App.js:4]: Why change from single to double quotes

File: src/Card.js [hunk 2]
```diff
@@ -3,74 +3,79 @@ import Avatar from "./Avatar";
...
-                    } }
-                >
-                        <Avatar picture={datum.actor}/>
-                        <Information data={datum}/>
-                </div>)
-        });
-        return result;
-            
-    };
+  componentWillReceiveProps(nextProps) {
```
Comment (abiodun0) [inline src/Card.js]: ComponentWillRecieveProps are being deprecated in the new react api, look at componentDidUpdate and tell me what you think

File: src/Card.js [hunk 3]
```diff
@@ -3,74 +3,79 @@ import Avatar from "./Avatar";
...
+        this.setState({ data });
+      })
+      .catch(error => this.setState({ error }));
+  };
 
-    checkError = (error) => {
-        return <p style={{padding: "10px", color: "red"}}>{error.message}</p>
-    }
+  checkError = error => {
+    return (
```
Comment (abiodun0) [inline src/Card.js:63]: Would this rather be a component on its own?

File: src/Information.js [hunk 4]
```diff
@@ -2,23 +2,22 @@ import React, { Component } from "react";
...
-            : ""}
-            </div>)
-    }
-};
-export default Information;
\ No newline at end of file
+  render() {
+    const { data } = this.props;
+    const type = data.type || "n/a";
+    const date = moment(data.created_at).toString() || "n/a";
```
Comment (abiodun0) [inline src/Information.js:8]: Why use moment here isn't this an over-kill?

File: src/test/App.test.js [hunk 5]
```diff
@@ -0,0 +1,8 @@
+import React from "react";
+import ReactDOM from "react-dom";
+import App from "../App";
+
+it("renders without crashing", () => {
```
Comment (abiodun0) [inline src/test/App.test.js:5]: What are you testing exactly here?

File: src/test/SearchTool.test.js [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+    expect(wrapper.toJSON()).toMatchSnapshot();
+  });
+
+  it("renders with props", () => {
+    // console.log(testInstance);
+    expect(testInstance.state).toEqual({ username: "aromokeye" });
+  });
+});
+
+// describe("SearchTool functionalities", () => {
```
Comment (abiodun0) [inline src/test/SearchTool.test.js]: Why did you leave commented out code here?

File: src/App.js [hunk 7]
```diff
@@ -1,35 +1,39 @@
...
+          background: "skyBlue",
+          border: "10px solid grey",
+          padding: "20px"
+        }}
       >
-        <SearchTool handleSearch={this.handleSearch}/>
-        { this.state.username.length? <GridView username={this.state.username} /> : "" }
+        <SearchTool handleSearch={this.handleSearch} />
+        {this.state.username.length ? (
+          <GridView username={this.state.username} />
```
Comment (abiodun0) [inline src/App.js:33]: Where is this gridView defined? + won't this be okay if you just use double && here?

File: src/App.js (hunk 1 above)
Comment (Seyi-Keye) [reply src/App.js:4]: That is the convention that my local development environment is set to work with. I use double quote for development

File: src/test/App.test.js (hunk 5 above)
Comment (Seyi-Keye) [reply src/test/App.test.js:5]: This is the default test that comes with CRA. It test that the app runs and the component renders succesfully.

File: src/Card.js (hunk 2 above)
Comment (Seyi-Keye) [reply src/Card.js]: ComponentDidUpdate works fine.

File: src/test/SearchTool.test.js (hunk 6 above)
Comment (Seyi-Keye) [reply src/test/SearchTool.test.js]: These are some edge cases that I am having problem testing. I wanted us to look at them together.

File: src/Information.js (hunk 4 above)
Comment (Seyi-Keye) [reply src/Information.js:8]: It was a very fast method of achieving my aim

File: src/Card.js (hunk 3 above)
Comment (Seyi-Keye) [reply src/Card.js:63]: I just needed it to be in the same container as the other Card displaying the result for simplicity.

File: src/App.js (hunk 7 above)
Comment (Seyi-Keye) [reply src/App.js:33]: I will look into it.

File: src/test/SearchTool.test.js (hunk 6 above)
Comment (Seyi-Keye) [reply src/test/SearchTool.test.js]: Similar cases in other component, I had issues with testing them.
